### PR TITLE
add created and updated at to AttackDetails

### LINKED ISSUE #33: Add CreatedAt and UpdatedAt fields to models.AttackDetails
Add the `CreatedAt` and `UpdatedAt` fields to `models.AttackDetails`, that get stored in the DB, to track the task submission time and the status update time.
The `CreatedAt` is set in `db.Add()` and the `UpdatedAt` is updated every time `db.Update()` is invoked with the `AttackDetails`.

### PR DESCRIPTION
**Pull request by enc**
If I read it correctly there is no mention of the AttackDetails in the call. Therefore no updated of the readme required.

resolves #33

### CODE REVIEW

**Comment by nitishm** [comment]
> If I read it correctly there is no mention of the AttackDetails in the call. Therefore no updated of the readme required.
> 
> resolves #33

This is my mistake. I should've mentioned `AttackInfo` and not `AttackDetails`. I want it to show up in the `GET /api/v1/attack(/<attackID>)` endpoint which returns `AttackResponse`, which is a typedef for `AttackInfo`.

Should be a quick change, to move it from `AttackDetails` to `AttackInfo` in the `models` package.

File: models/attack.go [hunk 1]
```diff
@@ -11,5 +13,7 @@ type AttackInfo struct {
 
 type AttackDetails struct {
 	AttackInfo
-	Result []byte `json:"result,omitempty"`
+	Result    []byte    `json:"result,omitempty"`
+	CreatedAt time.Time `json:"created_at"`
```
Comment (nitishm) [inline models/attack.go]: Move `CreatedAt` and `UpdatedAt` to `AttackInfo` struct above.

File: models/db.go [hunk 2]
```diff
@@ -73,6 +77,8 @@ func (tm TaskMap) Update(id string, attack AttackDetails) error {
 	}
 
 	mu.Lock()
+
```
Comment (nitishm) [inline models/db.go]: Nit - Remove newline.

**Comment by enc** [comment]
Updated the fields. I generally don't like to update fields levels down the hierarchy. But it keeps things simple.

We also have to recognize that the created-at field is having the semantics of a field on the AttackDetails but is found on the AttackInfo. slightly misleading but it fits well into the responses.

**Comment by nitishm** [comment]
Thanks for all the work @enc 🎉 !

**Pull request merged**